internal/rpm/rpm.go

```diff
@@ -12,21 +12,31 @@ import (
 	_ "github.com/glebarez/go-sqlite"
 )
 
-// GetPackageList returns the list of packages in the rpm database from either
-// /var/lib/rpm/rpmdb.sqlite, or /var/lib/rpm/Packages if the former does not exist.
+// GetPackageList returns the list of packages in the rpm database from
+// /var/lib/rpm/rpmdb.sqlite, /var/lib/rpm/Packages or /usr/lib/sysimage/rpm/rpmdb.sqlite.
 // If neither exists, this returns an error of type os.ErrNotExists
 func GetPackageList(ctx context.Context, basePath string) ([]*rpmdb.PackageInfo, error) {
-	rpmdirPath := filepath.Join(basePath, "var", "lib", "rpm")
-	rpmdbPath := filepath.Join(rpmdirPath, "rpmdb.sqlite")
-
-	if _, err := os.Stat(rpmdbPath); errors.Is(err, os.ErrNotExist) {
-		// rpmdb.sqlite doesn't exist. Fall back to Packages
-		rpmdbPath = filepath.Join(rpmdirPath, "Packages")
+	rpmdbPaths := []string{
+		// Explicitly check /usr/lib/sysimage/rpm. A compatibility symlink from
+		// /var/lib/rpm may not necessarily exist.
+		filepath.Join(basePath, "usr", "lib", "sysimage", "rpm", "rpmdb.sqlite"),
+		filepath.Join(basePath, "var", "lib", "rpm", "rpmdb.sqlite"),
+		filepath.Join(basePath, "var", "lib", "rpm", "Packages"),
+	}
 
-		// if the fall back path does not exist - this probably isn't a RHEL or UBI based image
-		if _, err := os.Stat(rpmdbPath); errors.Is(err, os.ErrNotExist) {
-			return nil, err
+	var rpmdbPath string
+	var errs []error
+	for _, path := range rpmdbPaths {
+		var err error
+		if _, err = os.Stat(path); err == nil {
+			rpmdbPath = path
+			break
 		}
+		errs = append(errs, err)
+	}
+
+	if rpmdbPath == "" {
+		return nil, errors.Join(errs...)
```

Comment:
This does still need the extra context that we had before. Just the arg to fmt.Errorf needed to change.